### ISSUE TITLE
Allow service containers to reached under a stable hostname

### ISSUE THREAD

**Issue by calind** [docker/docker#31119]
**Description**

I've seen this issue popping up around  various places, like #28861, #30353 and docker-library/rabbitmq#122 for example, but they are all closed. I've opened this as a feature request so that containers from the same service to be reachable using a stable hostname.

Currently you can set up service containers hostnames using templates like `{{.Task.Slot}}`, but they are not reachable using that hostname.

I think is important for statefull apps (eg. mysql) to have a stable hostname to reach their peers. Imagine a simple scenario with 2 mysql servers, `sql2` replicating from `sql1`. If `sql1` dies, it is respawned, but it's reachable under different hostname so `sql2` won't be able to automatically reconnect.

I think this scenario applies to a lot of clustered apps so lots of them would benefit from this feature.

**Comment by santiagopoli** [comment]
+1 for this feature! Although you can ping a service via its service name or its default hostname, there are some scenarios when picking a custom hostname might be useful. That can be done with templates, but as this issue states, there isn't currently a way to ping a service via a custom hostname set with templates.

**Comment by cpuguy83** [comment]
I agree, though `{{ .Task.Slot }}` is not necessarily stable either, IIUC.

**Comment by SAprelov** [comment]
Any news?
